### PR TITLE
Private/jameshen/host https only

### PR DESCRIPTION
**Pull request by jameshensmancitrix**
Added https_only field to host as well as the setter

### CODE REVIEW

File: ocaml/idl/datamodel_host.ml [hunk 1]
```diff
@@ -1727,6 +1727,16 @@ let copy_primary_host_certs =
       ]
     ~allowed_roles:_R_POOL_ADMIN ~hide_from_docs:true ~pool_internal:true ()
 
+let set_https_only =
+  call ~name:"set_https_only" ~doc:"updates the host firewall to open or close port 80 depending on the value" 
+  ~lifecycle:[]
+  ~params:
+  [
+      (Ref _host, "host", "the host that the https_only setting is to be set")
+    ; (Bool, "value", "host.https_only will be updated according to this value")
```
Comment (lindig) [inline ocaml/idl/datamodel_host.ml:1736]: I think the doc-strings could be less self-referential: "the host that the https_only setting is to be set" -> "host whose firewall port settings to update"; "host.https_only will be updated according to this value" -> "Boolean for https_only; true - http port 80 will be blocked, false - http port 80 will be open".

File: ocaml/xapi-cli-server/records.ml [hunk 2]
```diff
@@ -2968,6 +2968,7 @@ let host_record rpc session_id host =
           ()
       ; make_field ~name:"hostname" ~get:(fun () -> (x ()).API.host_hostname) ()
       ; make_field ~name:"address" ~get:(fun () -> (x ()).API.host_address) ()
+      ; make_field ~name:"https_only" ~get:(fun () -> string_of_bool (x ()).API.host_https_only) ~set:(fun s -> Client.Host.set_https_only ~rpc ~session_id ~host ~value:(Stdlib.bool_of_string s) ) ()
```
Comment (lindig) [inline ocaml/xapi-cli-server/records.ml:2971]: Are these lines not folded?

**Review by lindig — APPROVED** [review]

File: ocaml/idl/datamodel_host.ml [hunk 3]
```diff
@@ -1926,6 +1937,7 @@ let t =
             ~doc_tags:[Networking] "address"
             "The address by which this host can be contacted from any other \
              host in the pool"
+        ; field ~qualifier:DynamicRO ~ty:Bool ~default_value:(Some (VBool false)) "https_only" "Reflets wheather port 80 is open (false) or not (true)"
```
Comment (robhoes) [inline ocaml/idl/datamodel_host.ml:1940]: Please add new fields to the end of the list. Also there are spelling mistakes in the doc string.

Comment (robhoes) [reply ocaml/idl/datamodel_host.ml:1940]: I think you also need to add `~lifecycle:[]` here to get datamodel_lifecycle.ml updated properly.

File: ocaml/idl/datamodel_host.ml [hunk 4]
```diff
@@ -1727,6 +1727,16 @@ let copy_primary_host_certs =
       ]
     ~allowed_roles:_R_POOL_ADMIN ~hide_from_docs:true ~pool_internal:true ()
 
+let set_https_only =
+  call ~name:"set_https_only" ~doc:"updates the host firewall to open or close port 80 depending on the value" 
+  ~lifecycle:[]
+  ~params:
+  [
+      (Ref _host, "host", "the host that the https_only setting is to be set")
```
Comment (robhoes) [inline ocaml/idl/datamodel_host.ml:1735]: The param name should be `self`, not `host`. For the doc string, probably just have "The host".

File: ocaml/xapi-cli-server/records.ml (hunk 2 above)
Comment (robhoes) [reply ocaml/xapi-cli-server/records.ml:2971]: Indeed, `make format` should be used.

File: ocaml/xapi/message_forwarding.ml [hunk 5]
```diff
@@ -3961,6 +3961,11 @@ functor
         let uuid = host_uuid ~__context self in
         info "Host.apply_updates: host = '%s'; hash = '%s'" uuid hash ;
         Local.Host.apply_updates ~__context ~self ~hash
+
+      let set_https_only ~__context ~host ~value = 
+        info "setting https_only to %B" value;
```
Comment (robhoes) [inline ocaml/xapi/message_forwarding.ml:3966]:
```suggestion
        info "Host.set_https_only: self = '%s';  value = %b" self value ;
```
(as per convention)

File: ocaml/xapi-cli-server/records.ml [hunk 6]
```diff
@@ -2968,6 +2968,13 @@ let host_record rpc session_id host =
           ()
       ; make_field ~name:"hostname" ~get:(fun () -> (x ()).API.host_hostname) ()
       ; make_field ~name:"address" ~get:(fun () -> (x ()).API.host_address) ()
+      ; make_field ~name:"https_only"
```
Comment (robhoes) [inline ocaml/xapi-cli-server/records.ml:2971]:
```suggestion
      ; make_field ~name:"https-only"
```

File: ocaml/xapi/message_forwarding.ml (hunk 5 above)
Comment (psafont) [reply ocaml/xapi/message_forwarding.ml]: Even more handily:
```suggestion
        info "%s: self = '%s';  value = %b" __FUNCTION__ self value ;
```

File: ocaml/idl/datamodel_host.ml [hunk 7]
```diff
@@ -2077,6 +2095,9 @@ let t =
             "last_software_update"
             ~default_value:(Some (VDateTime (Date.of_float 0.0)))
             "Date and time when the last software update was applied"
+        ; field ~qualifier:DynamicRO ~lifecycle:[] ~ty:Bool
+            ~default_value:(Some (VBool false)) "https_only"
+            "Reflets wheather port 80 is open (false) or not (true)"
```
Comment (psafont) [inline ocaml/idl/datamodel_host.ml:2100]:
```suggestion
            "Reflects whether port 80 is open (false) or not (true)"
```

File: ocaml/idl/datamodel_host.ml [hunk 8]
```diff
@@ -1727,6 +1727,23 @@ let copy_primary_host_certs =
...
+      "updates the host firewall to open or close port 80 depending on the \
+       value"
+    ~lifecycle:[]
+    ~params:
+      [
+        (Ref _host, "self", "The Host")
+      ; ( Bool
+        , "value"
+        , "Boolean for https_only; true - http port 80 will be blocked, false \
+           - http port 80 will be open"
```
Comment (psafont) [inline ocaml/idl/datamodel_host.ml:1742]: I would remove the first sentence altogether, that information is already displayed elsewhere (it's a boolean by the type and the value controls by its name)

```suggestion
        , "true - blocks http port 80; false - opens http port 80"
```

File: datamodel_lifecycle.ml [hunk 9]
```diff
@@ -0,0 +1,43 @@
+let prototyped_of_class = function
```
Comment (robhoes) [inline datamodel_lifecycle.ml:1]: This one came back...

Comment (robhoes) [reply datamodel_lifecycle.ml]: and gone :)

**Review by robhoes — APPROVED** [review]

**Pull request merged**